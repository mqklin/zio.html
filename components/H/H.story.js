import React from 'react';
import { storiesOf } from '@storybook/react';

import  H  from './index';

const customSize = {
  fontSize:"59px",
  color:"purple",
  transform: "rotate(7deg)",
  margin: "100px",
  display: "inline-block",
  border: "5px solid cyan"
}

storiesOf('H', module)
  .add('Size | xs', () => <H size="xs">Very Small 😊</H>)
  .add('Size | sm', () => <H size="sm">Just Small 😊</H>)
  .add('Size | md', () => <H size="md">Medium Size 😊</H>)
  .add('Size | lg', () => <H size="lg">Big Size 😁</H>)
  .add('Size | xl', () => <H size="xl">Veryy Big Size 😱</H>)
  .add('Inner Tag', () => <H size="md"><a href="#"> Send me nowhere</a></H>)
  .add('Custom Style', () => <H extraStyle={customSize} size="md">Its Custom Style</H>)
