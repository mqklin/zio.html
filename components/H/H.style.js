import { Default } from "../Default/default.style";
const DefaultStyle = Default.withComponent("h3");

const sizes = {
  xs: "0.67em",
  sm: "0.83em",
  md: "1.17em",
  lg: "1.5em",
  xl: "2em"
};

export const HTag = DefaultStyle.extend`
  font-size: ${props =>
    props.size
      ? sizes[props.size]
      : props.fontSize ? props.fontSize : sizes.md};
  text-transform: ${props => props.textTransform || "none"};
  display: ${props => props.display || "block"};
`;
