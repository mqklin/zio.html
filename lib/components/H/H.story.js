'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

require('./story/custom.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

(0, _react3.storiesOf)('H', module).add('Size | xs', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'xs' },
    'Very Small \uD83D\uDE0A'
  );
}).add('Size | sm', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'sm' },
    'Just Small \uD83D\uDE0A'
  );
}).add('Size | md', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'md' },
    'Medium Size \uD83D\uDE0A'
  );
}).add('Size | lg', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'lg' },
    'Big Size \uD83D\uDE01'
  );
}).add('Size | xl', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'xl' },
    'Veryy Big Size \uD83D\uDE31'
  );
}).add('Inner Tag', function () {
  return _react2.default.createElement(
    _index2.default,
    { size: 'md' },
    _react2.default.createElement(
      'a',
      { href: '#' },
      ' Send me nowhere'
    )
  );
}).add('Custom Style', function () {
  return _react2.default.createElement(
    _index2.default,
    { className: 'custom-component', size: 'md' },
    'Its Custom Style'
  );
});