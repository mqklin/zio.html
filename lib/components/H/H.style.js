"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HTag = undefined;

var _templateObject = _taggedTemplateLiteral(["\n  font-size: ", ";\n"], ["\n  font-size: ", ";\n"]);

var _styledComponents = require("styled-components");

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var sizes = {
  xs: "0.67em",
  sm: "0.83em",
  md: "1.17em",
  lg: "1.5em",
  xl: "2em"
};

var HTag = exports.HTag = _styledComponents2.default.h3(_templateObject, function (props) {
  return props.size ? sizes[props.size] : props.fontSize ? props.fontSize : sizes.md;
});