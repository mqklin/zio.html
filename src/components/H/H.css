.zio_html-h {
  display: block;
}
