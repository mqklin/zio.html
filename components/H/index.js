import React, { Component } from 'react';
import PropTypes from "prop-types";
import { HTag } from "./H.style";

/**
 * The HTML H1,H2,H3,H4,H5,H6
 */
class H extends Component {
  render() {
    const { children:innerHTML,extraStyle,size } = this.props;
    return (
      <HTag style={extraStyle} size={size}>
        {innerHTML}
      </HTag>
    )
  }
}

H.propTypes = {
  /**
  * Extra Tag Styling
  */
  extraStyle:PropTypes.object,
  /**
  * The size of font size
  * xs: 0.67em
  * sm: 0.83em
  * md: 1.17em
  * lg: 1.5em
  * xl: 2em
  */
  size:PropTypes.oneOf(['xs','sm','md','lg','xl']).isRequired,
  /**
  * The innerHTML what would be inside the tag text or another Tag *Better not to use another tag*
  */
  children: PropTypes.node.isRequired
};


export default H;
