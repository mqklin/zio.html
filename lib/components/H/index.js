"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _H = require("./H.style");

require("./H.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * The HTML H1,H2,H3,H4,H5,H6,
 * @param {Object} extraStyle  Extra Tag Styling using styled-components
 * @param {string} className  Extra Tag Styling using css
 * @param {string} size The size of font size
 * xs: 0.67em
 * sm: 0.83em
 * md: 1.17em
 * lg: 1.5em
 * xl: 2em
 * @param {mix} children The innerHTML what would be inside the tag text or another Tag *Better not to use another tag*
 */
var H = function (_Component) {
  _inherits(H, _Component);

  function H() {
    _classCallCheck(this, H);

    return _possibleConstructorReturn(this, (H.__proto__ || Object.getPrototypeOf(H)).apply(this, arguments));
  }

  _createClass(H, [{
    key: "render",
    value: function render() {
      var _props = this.props,
          innerHTML = _props.children,
          extraStyle = _props.extraStyle,
          size = _props.size,
          externalStylingClass = _props.className;

      return _react2.default.createElement(
        _H.HTag,
        { style: extraStyle, className: "zio_html-h " + externalStylingClass, size: size },
        innerHTML
      );
    }
  }]);

  return H;
}(_react.Component);

H.propTypes = {
  className: _propTypes.string,
  extraStyle: _propTypes.object,
  size: (0, _propTypes.oneOf)(['xs', 'sm', 'md', 'lg', 'xl']),
  children: _propTypes.node.isRequired
};
exports.default = H;