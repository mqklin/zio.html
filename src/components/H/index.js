import React, { Component } from 'react';
import { string,object, oneOf,node } from "prop-types";
import { HTag } from "./H.style";
import './H.css';


/**
 * The HTML H1,H2,H3,H4,H5,H6,
 * @param {Object} extraStyle  Extra Tag Styling using styled-components
 * @param {string} className  Extra Tag Styling using css
 * @param {string} size The size of font size
 * xs: 0.67em
 * sm: 0.83em
 * md: 1.17em
 * lg: 1.5em
 * xl: 2em
 * @param {mix} children The innerHTML what would be inside the tag text or another Tag *Better not to use another tag*
 */
class H extends Component {
  static propTypes = {
    className: string,
    extraStyle: object,
    size: oneOf(['xs','sm','md','lg','xl']),
    children: node.isRequired
  }
  render() {
    const {
      children:innerHTML,
      extraStyle,
      size,
      className:externalStylingClass
    } = this.props;
    return (
      <HTag style={extraStyle} className={`zio_html-h ${ externalStylingClass }`} size={size}>
        {innerHTML}
      </HTag>
    )
  }
}

export default H;
