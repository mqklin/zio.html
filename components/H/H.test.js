import React from "react";
import { mount } from "enzyme";
import { H } from "./H";

describe("H", () => {
  describe("props", () => {
    it("should change the text of tag", () => {
      const wrapper = mount(<H>unitest</H>);
      expect(wrapper.text()).toBe("unitest");

      wrapper.setProps({ children: "newtext" });
      expect(wrapper.text()).toBe("newtext");
    });
  });
});
