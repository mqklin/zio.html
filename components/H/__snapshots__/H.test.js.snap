// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H should be equal to snapshow 1`] = `
<h3
  className="sc-htpNat gWWnZS"
  size="xs"
  style={undefined}
>
  I am small ðŸ‘Œ
</h3>
`;
